dvc/repo/__init__.py

```diff
@@ -257,17 +257,19 @@ def used_cache(
         cache["azure"] = []
 
         for branch in self.scm.brancher(
-            all_branches=all_branches, all_tags=all_tags
+            all_branches=all_branches, all_tags=all_tags, checkout=False
```

Comment:
Could we make `scm` decide if we do checkout or not? What was the reason to introduce this additional parameter?

Reply:
The reason - to keep the same behavior as it was without it. `brancher()` could be useful if we want to introduce an internal API like `git filter-branch` tree filters [again].

Reply:
if we don't use it now I would remove it. Maintaining unused code creates additional burden - writing and maintaining tests, additional complexity when you do some refactoring, cognitive complexity.